lib/wasix/src/syscalls/wasix/proc_join.rs

```diff
@@ -182,18 +182,30 @@ pub(super) fn proc_join_internal<M: MemorySize + 'static>(
             }
         ));
 
-        // Wait for the process to finish
-        let process2 = process.clone();
-        let res =
-            __asyncify_with_deep_sleep::<M, _, _>(ctx, Duration::from_millis(50), async move {
-                let exit_code = process.join().await.unwrap_or_else(|_| Errno::Child.into());
-                tracing::trace!(%exit_code, "triggered child join");
-                JoinStatusResult::ExitNormal(pid, exit_code)
-            })?;
-        return match res {
-            AsyncifyAction::Finish(ctx, result) => ret_result(ctx, result),
-            AsyncifyAction::Unwind => Ok(Errno::Success),
-        };
+        if flags & JoinFlags::NON_BLOCKING != JoinFlags::from_bits_preserve(0) {
+            return if let Some(status) = process.try_join() {
```

Comment:
I know this was already in the code before, but return statements before large expressions are very easy to miss. 

Also, unless I'm missing something, it seems like this code will always hit one of the two branches, right?

So we can just let the value fall through and remove the `Nothing` at the end?

Reply:
Unfortunately it was nested in one more outer if statement: https://github.com/wasmerio/wasmer/pull/4202/files#diff-26f88335b0c3e9515dca019801949f26b612efb713b4182b1b23a80d6a2d6dcaR175 . I changed it to if/else to remove the "returns".